lib/services/credentials/credential.service.js

```diff
@@ -45,84 +45,94 @@ s.insertCredential = function (id, type, credentialDetails) {
     throw new Error(`Invalid credential type: ${type}`); // TODO: replace with validation error
   }
 
-  return this.getCredential(id, type) // check if credential already exists
-    .then((cred) => {
-      if (cred && cred.isActive) {
-        throw new Error('Credential already exists and is active'); // TODO: replace with validation error
-      }
-      return dereferencePromise;
-    })
-    .then((resolvedSchema) => {
-      const credentialConfig = resolvedSchema.properties[type];
-      const newCredential = { isActive: 'true' };
-      utils.appendCreatedAt(newCredential);
-      utils.appendUpdatedAt(newCredential);
+  // TODO: not a good approach, new way TBD
+  const areMultipleCredsAllowed = ['key-auth', 'jwt'].includes(type);
+
+  let credentialExistsPromise;
+  if (areMultipleCredsAllowed) {
```

Comment:
I think we can restructure the flow to be a little bit cleaner.

We could extract the checking function so that we can do (pseudo code)

```js
const checkSingleCredExistence = () => {};

const flow = areMultipleCredsAllowed ? 
  dereferencePromise.then(checkSingleCredExistence) : 
  dereferencePromise 
```

Reply:
@XVincentX do you mean something like this?

  ```javascript
  // check if credential already exists
  const checkSingleCredExistence = () => {
    return this.getCredential(id, type)
      .then(cred => {
        if (cred && cred.isActive) {
          throw new Error('Credential already exists and is active'); // TODO: replace with validation error
        }
      });
  };

  // TODO: not a good approach, new way TBD
  const areMultipleCredsAllowed = ['key-auth', 'jwt'].includes(type);
  const flow = areMultipleCredsAllowed ?
    dereferencePromise :
    checkSingleCredExistence().then(() => dereferencePromise);

  return flow.then(resolvedSchema => {
    ...
  });
  ```

Reply:
Yeah exactly!